### ISSUE TITLE
Let players attack Monsters/Mobs in claims/subclaims

### ISSUE THREAD

**Issue by BushGamingYT** [ShaneBeee/GriefPreventionFlags#68]
### Request
Could you add a flag that allows players to kill mobs and/or monsters inside of a claim/sub-claim but won't let players kill each other and destroy the claim? (btw, when I say mobs, I mean animals and such (non-harmful creatures)).
### Use
This can be used in many ways especially for sub-claims like Grinders, Experience gaining, McMMO upgrades and the way I would like to use it (On my private server) is to have a sub-claim in spawn that has pig-spawners and lets players kill the pigs without being able to break the surrounding area or kill each other. I can see this helping many servers that don't want to or don't know how to use world guard. 
### Commands
               Command                                                       Description
     /SetClaimFlag PlayerKillsMobs                   Enables Players to kill mobs within that claimed area
     /SetClaimFlag PlayerKillsMobType [Type]         Enables Players to kill a type of mob withing the claimed area
     /SetClaimFlag PlayerKillsMonsters               Enables Players to kill monsters within that claimed area
     /SetClaimFlag PlayerKillsMonsterType [Type]     Enables Players to kill a type of monster within that claimed area

**Comment by ShaneBeee** [comment]
I'm not sure if this will be possible.
GP prevents players from hurting animals, and I don't see an event from GP to cancel to be able to override this.

**Comment by BushGamingYT** [comment]
Is there not a way to see what is actually stopping the player from killing the mob and then block it or sense when it blocks the murder of the animal then your plugin allows the player to kill the animal?. There is obviously something stopping the player from killing animal otherwise it would not be blocked.

**Comment by ShaneBeee** [comment]
Yes, I could probably grab the original event and un-cancel it. 
This comes with its own issues though as it may interfere with other plugins cancelling the same event.

There is no purpose adding a flag for something that can easily be done using GP

**Comment by BushGamingYT** [comment]
The reasons I think this should at least be tried to be added is that
1) Container trust, as far as I know, does not allow the killing of monsters
2) If someone was making a professional server and had people coming in over time and was not on when they joined or leaved then they would not have access to the animals (especially when they need to kill the animals of start for food)
3) It will only let them kill mobs and not access chests

**Comment by ShaneBeee** [comment]
1) You can kill monsters in claims without any permissions. Im assuming this was setup in the case the monster tries to attack you, you should be able to fight back. (tested and it works)
2) You can add trust to public `/containertrust public` which will give access to all players.
3) You can create a subclaim around the area that players need to kill mobs (ie: around the grinder)
4) I get where you are coming from, but its a lot of work to add a flag, and you are requesting 4 flags. On top of that, with the exception of being able to choose the entity types, this can all be done via GP without the need for flags.

In my opinion these flags are a little redundant, as I stated already this can be handled already in GP without the need for the flags. 

I'll leave this issue open and mark it as an enchantment, if someone wants to do the work and PR it, I may accept/pull it. I just don't feel like adding a flag that serves no purpose.

**Comment by BushGamingYT** [comment]
Ok, thank you for telling me this information. Sorry I did not know about this before (I last properally used go about a year or 2 ago), also thank you for considering to add it or let others have a try. I will use the gp command but if you ever do add the commands, that would be amazing. I know coding is hard and my first ever proper plugin took ages but seems simple, so whenever I ask for a suggestion on a plugin, I take in the fact that they don't have to do it for free and choose to give it to the public which is awesome.
